frontend/src/components/pages/UserList.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+
+const UserList = () => {
+  const [users, setUsers] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  const fetchUsers = async () => {
+    const response = await axios.get('http://127.0.0.1:8000/api/user-list');
+
+    if (response.data.status === 200) {
+      setUsers(response.data.users);
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
+
+  const userlist_HTMLTABLE = users.map((user) => (
+    <tr key={user.id}>
+      <td data-label='Username'>
+        <Link to={`/user-profile/${user.username}`}>{user.username}</Link>
+      </td>
+      <td data-label='Name'>{user.name}</td>
+      <td data-label='Email'>{user.email}</td>
+    </tr>
+  ));
```

Comment:
kani sya na function equivalent na ni siya as a React component. Since React components are just function too. You can call it like this

`<userlist_HTMLTABLE />`

But since it not nice to look at like that, so rename it and apply studly case `StudlyCase`.

`<UserList />`

What's more organize is that this are called "Dummy Components" meaning these components does not have states just render. So please create a new file for it.